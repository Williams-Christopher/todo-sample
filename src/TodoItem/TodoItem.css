.todolist_item {
    border: 1px solid brown;
    text-align: left;
    margin: 15px auto 0 auto;
    padding: 10px;
    width: 80%;
}

.todolist_content {

}

.todolist_date {
    font-style: italic;
}
