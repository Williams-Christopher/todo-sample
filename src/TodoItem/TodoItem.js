import React from 'react';

import './TodoItem.css';

function TodoItem(props) {
    return (
        <article className='todolist_item'>
            <p className='todolist_title'>{props.title}</p>
            <p className='todolist_date'>Date added: <time>{props.dateAdded}</time></p>
            <p className='todolist_date'>Date due: <time>{props.dateDue}</time></p>
            <p className='todoitem_note'>{props.note}</p>
            <button type='button' onClick={e => props.deleteItem(props.id)}>Delete</button>
        </article>
    )
}

TodoItem.defaultProps = {
    id: '',
    title: '',
    dateAdded: '',
    dateDue: '',
    note: '',
    deleteItem: () => {}
}

export default TodoItem;
