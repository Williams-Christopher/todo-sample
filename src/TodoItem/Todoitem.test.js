import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TodoItem from './TodoItem';

const props = {
    id: '123456789',
    title: 'Test item',
    note: 'This is a test note',
    dateAdded: '2019-01-01T00:00:00Z',
    dateDue: '2019-01-02T23:59:59Z'
}

it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TodoItem />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render with content as expected', () => {
    const tree = renderer.create(<TodoItem id={props.id} title={props.title} note={props.note} dateAdded={props.dateAdded} dateDue={props.dateDue} />).toJSON();
    expect(tree).toMatchSnapshot();
})