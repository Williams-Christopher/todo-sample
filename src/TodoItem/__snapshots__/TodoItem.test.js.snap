// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with content as expected 1`] = `
<article
  className="todolist_item"
>
  <p
    className="todolist_title"
  >
    Test item
  </p>
  <p
    className="todolist_date"
  >
    Date added: 
    <time>
      2019-01-01T00:00:00Z
    </time>
  </p>
  <p
    className="todolist_date"
  >
    Date due: 
    <time>
      2019-01-02T23:59:59Z
    </time>
  </p>
  <p
    className="todoitem_note"
  >
    This is a test note
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</article>
`;
